### PR TITLE
Better filter generators and loads at GLSK import

### PR DESCRIPTION
**Pull request by murgeyseb**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature modification

**What is the current behavior?** *(You can also link to an open issue here)*
Currently, GLSK importer does not check that an injection is connected to the main synchronous component to add it to the GLSK.

**What is the new behavior (if this is a feature change)?**
GLSK importer does check that an injection is connected to the main synchronous component before adding it to the generated Scalable object.

### CODE REVIEW

File: data/glsk/glsk-file/src/main/java/com/farao_community/farao/data/glsk/import_/actors/GlskPointScalableConverter.java [hunk 1]
```diff
@@ -120,23 +120,26 @@ private static void convertCountryProportional(Network network, GlskShiftKey gls
 
         if (glskShiftKey.getPsrType().equals("A04")) {
             LOGGER.debug("GLSK Type B42, empty registered resources list --> country (proportional) GSK");
+            List<Generator> generators = network.getGeneratorStream().filter(generator -> country.equals(generator.getTerminal().getVoltageLevel().getSubstation().getCountry().orElse(null)))
```
Comment (alexandrem-rte) [inline data/glsk/glsk-file/src/main/java/com/farao_community/farao/data/glsk/import_/actors/GlskPointScalableConverter.java]:
```suggestion
            List<Generator> generators = network.getGeneratorStream()
                .filter(generator -> generator.getTerminal().getVoltageLevel().getSubstation().getCountry().orElse(null)).equals(country))
```

File: data/glsk/glsk-file/src/main/java/com/farao_community/farao/data/glsk/import_/actors/GlskPointScalableConverter.java [hunk 2]
```diff
@@ -120,23 +120,26 @@ private static void convertCountryProportional(Network network, GlskShiftKey gls
...
-            network.getGeneratorStream().filter(generator -> generator.getTerminal().getVoltageLevel().getSubstation().getCountry().orElse(null).equals(country))
-                    .forEach(generator -> percentages.add(100 * glskShiftKey.getQuantity().floatValue() * (float) generator.getTargetP() / (float) totalCountryP));
-            network.getGeneratorStream().filter(generator -> generator.getTerminal().getVoltageLevel().getSubstation().getCountry().orElse(null).equals(country))
-                    .forEach(generator -> scalables.add(Scalable.onGenerator(generator.getId())));
+            generators.forEach(generator -> percentages.add(100 * glskShiftKey.getQuantity().floatValue() * (float) generator.getTargetP() / (float) totalCountryP));
+            generators.forEach(generator -> scalables.add(Scalable.onGenerator(generator.getId())));
         } else if (glskShiftKey.getPsrType().equals("A05")) {
             LOGGER.debug("GLSK Type B42, empty registered resources list --> country (proportional) LSK");
+            List<Load> loads = network.getLoadStream().filter(load -> country.equals(load.getTerminal().getVoltageLevel().getSubstation().getCountry().orElse(null)))
```
Comment (alexandrem-rte) [inline data/glsk/glsk-file/src/main/java/com/farao_community/farao/data/glsk/import_/actors/GlskPointScalableConverter.java]:
```suggestion
            List<Load> loads = network.getLoadStream().filter(load ->
                load.getTerminal().getVoltageLevel().getSubstation().getCountry().orElse(null)).equals(country))
```

File: data/glsk/glsk-file/src/main/java/com/farao_community/farao/data/glsk/import_/actors/GlskPointScalableConverter.java [hunk 3]
```diff
@@ -150,21 +153,31 @@ private static void convertCountryProportional(Network network, GlskShiftKey gls
     private static void convertExplicitProportional(Network network, GlskShiftKey glskShiftKey, List<Float> percentages, List<Scalable> scalables, TypeGlskFile typeGlskFile) {
         if (glskShiftKey.getPsrType().equals("A04")) {
             LOGGER.debug("GLSK Type B42, not empty registered resources list --> (explicit/manual) proportional GSK");
-            List<String> genenratorsList = glskShiftKey.getRegisteredResourceArrayList().stream().map(generatorResource -> generatorResource.getGeneratorId(typeGlskFile)).collect(Collectors.toList());
-            double totalP = network.getGeneratorStream().filter(generator -> genenratorsList.contains(generator.getId())).mapToDouble(Generator::getTargetP).sum();
+            List<Generator> generatorsList = glskShiftKey.getRegisteredResourceArrayList().stream()
```
Comment (alexandrem-rte) [inline data/glsk/glsk-file/src/main/java/com/farao_community/farao/data/glsk/import_/actors/GlskPointScalableConverter.java]: rename generatorsList to generators

File: data/glsk/glsk-file/src/main/java/com/farao_community/farao/data/glsk/import_/actors/GlskPointScalableConverter.java [hunk 4]
```diff
@@ -176,28 +189,31 @@ private static void convertExplicitProportional(Network network, GlskShiftKey gl
      * @param scalables list of scalable
      */
```
Comment (alexandrem-rte) [inline data/glsk/glsk-file/src/main/java/com/farao_community/farao/data/glsk/import_/actors/GlskPointScalableConverter.java]: missing parameter in the docstring

File: data/glsk/glsk-file/src/main/java/com/farao_community/farao/data/glsk/import_/actors/GlskPointScalableConverter.java [hunk 5]
```diff
@@ -176,28 +189,31 @@ private static void convertExplicitProportional(Network network, GlskShiftKey gl
...
     private static void convertParticipationFactor(Network network, GlskShiftKey glskShiftKey, List<Float> percentages, List<Scalable> scalables, TypeGlskFile typeGlskFile) {
-        List<GlskRegisteredResource> resourceList = glskShiftKey.getRegisteredResourceArrayList();
-
         if (glskShiftKey.getPsrType().equals("A04")) {
             LOGGER.debug("GLSK Type B43 GSK");
-            double totalFactor = glskShiftKey.getRegisteredResourceArrayList().stream()
+            List<GlskRegisteredResource> generatorsList = glskShiftKey.getRegisteredResourceArrayList().stream()
```
Comment (alexandrem-rte) [inline data/glsk/glsk-file/src/main/java/com/farao_community/farao/data/glsk/import_/actors/GlskPointScalableConverter.java]: rename to generators?

**Comment by murgeyseb** [comment]
> I made a few comments/suggestions in the code.
> Do the existing tests cover the new features? Wouldn't it be useful to add a log saying that some generators/loads included in the GLSK file have been filtered because they were not in the main synchronous component?

For this we have Quality Check module, so I do not thik it is so important yet.

**Pull request merged**